utils/textmate/Syntaxes/Carbon.plist

```diff
@@ -143,7 +143,7 @@ SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
                 <key>name</key>
                 <string>constant.character.escape.carbon</string>
                 <key>match</key>
-                <string>\\([tnr'"0\0]|x[0-9A-F]{2}|u\{[0-9A-F]{4,}\})</string>
+                <string>\\('''|[tnr'"0\0]|x[0-9A-F]{2}|u\{[0-9A-F]{4,}\})</string>
```

Comment:
I would have expected this is fine to just have with the single escaped `'`? The escape sequence is just `\'`, and it can then be followed by two `'`s without forming the close `'''`.... At least, that's my reading of this section:
https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/lexical_conventions/string_literals.md#:~:text=whose%20first%20%27%20is%20not%20part%20of%20a%20%5C%27%20escape%20sequence

Or does that result in VS Code still thinking the block literal is closed?

Reply:
Single escaped `'` results in only the first `'` of `'''` getting colored (look at #3634, it's visible in last image). Also single `'` is already covered by the second capturing group of this regex.

Reply:
> Single escaped `'` results in only the first `'` of `'''` getting colored (look at #3634, it's visible in last image).

I think that's correct though? There is one escaped character (the first `'`), and two normal characters in the body of the block string literal.

Here is an example to illustrate what I'm talking about maybe:

```carbon
let example: auto = '''
    One and two quotes are fine w/o escaping: ' ''
    Can also be escaped: \' \'\'
    First quote escaped makes the rest normal, and not the end:
      \'''
    Could also escape all three:
      \'\'\'
    But without any escapes, we end the string with three:
    ''';
```

> Also single `'` is already covered by the second capturing group of this regex.

Yeah, and I think that's probably all we need here?

Reply:
> First quote escaped makes the rest normal, and not the end:
      \'''

Ok, I see.. I'll revert this part.